src/ui/hash.js

```diff
@@ -67,19 +70,47 @@ class Hash {
         if (mapFeedback) {
             // new map feedback site has some constraints that don't allow
             // us to use the same hash format as we do for the Map hash option.
-            hash += `#/${lng}/${lat}/${zoom}`;
+            hash += `/${lng}/${lat}/${zoom}`;
         } else {
-            hash += `#${zoom}/${lat}/${lng}`;
+            hash += `${zoom}/${lat}/${lng}`;
         }
 
         if (bearing || pitch) hash += (`/${Math.round(bearing * 10) / 10}`);
         if (pitch) hash += (`/${Math.round(pitch)}`);
-        return hash;
+
+        if (this._hashName && !mapFeedback) {
+            let found = false;
+            const parts = window.location.hash.slice(1).split('&').map(part => {
+                const key = part.split('=')[0];
+                if (key === this._hashName) {
+                    found = true;
+                    return `${key}=${hash}`;
+                }
+                return part;
+            }).filter(a => a);
+            if (!found) {
+                parts.push(`${this._hashName || ''}=${hash}`);
```

Comment:
Do we need the `|| ''` part here? If there's no hash name, then shouldn't we skip the `=` as well?

Reply:
`test-flow` fails without this, see https://circleci.com/gh/mapbox/mapbox-gl-js/50268
```
Cannot coerce `this._hashName` to string because null or undefined [1] should not be coerced.

   src/ui/hash.js:92:31
   92|                 parts.push(`${this._hashName}=${hash}`);
                                     ^^^^^^^^^^^^^^
```

Reply:
We can get around this by assigning `this._hashName` to a variable at the top of the if statement and then changing the two references to `this._hashName` to the variable. That lets us get rid of the unnecessary and confusing `|| ''` part of this while keeping Flow happy.